# Evaluation Regression Models
- Tahmin sonucunun ne kadar doğru olduğunu gösterir.
- residual = y - y_head
- Artı ve eksi değerleri sonucu etkilememsi için = (residual)^2
- sum square residual = sum((y-y_head)^2)
- Burası benim ilk ihtiyacım olan sayısal değer. Sonra burada ki y ekseninde bir değeri var. Ben bunların ortalmasını buluyorum. Ortalama değer örneğin => y_avg = 12000
- Şöyle bir formul uyarlıyorum => sum square total = sum((y-y_avg)^2)
- Bir değerlendirme metriği var. Yani modelimizi değerlendireceğimiz sayısal sonuç. <br>
  => R^2 = 1 - (SSR / SST) <br>
  SSR = sum square residual <br>
  SST = sum square total <br>
  R^2 değeri 1'e ne kadar yakın ise bu o kadar iyi demektir. <br>
  R^2 = r_score

In [17]:
import pandas as pd
import numpy as np
import matplotlib as np

df = pd.read_csv("descisionTreeDataset.csv",sep=";",header=None)
df

0    1
0   1  100
1   2   80
2   3   70
3   4   60
4   5   50
5   6   40
6   7   30
7   8   20
8   9   10
9  10    5

In [20]:
x = df.iloc[:,0].values.reshape(-1,1)
y = df.iloc[:,1].values.reshape(-1,1)

from sklearn.ensemble import RandomForestRegressor
rf = RandomForestRegressor(n_estimators=100, random_state=42)
rf.fit(x,y)

C:\Users\mfurk\Anaconda3\lib\site-packages\ipykernel_launcher.py:6: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples,), for example using ravel().
  


RandomForestRegressor(bootstrap=True, criterion='mse', max_depth=None,
                      max_features='auto', max_leaf_nodes=None,
                      min_impurity_decrease=0.0, min_impurity_split=None,
                      min_samples_leaf=1, min_samples_split=2,
                      min_weight_fraction_leaf=0.0, n_estimators=100,
                      n_jobs=None, oob_score=False, random_state=42, verbose=0,
                      warm_start=False)

In [28]:
y_head = rf.predict(x)
y_head

array([89.1 , 83.3 , 72.1 , 61.9 , 52.1 , 42.8 , 30.8 , 22.7 , 12.65,
        8.5 ])

In [29]:
from sklearn.metrics import r2_score
print("r_score:",r2_score(y,y_head))

r_score: 0.9798724794092587


- 0.98'e  bir değer ile 1'e yakın çıkmıştır. Yani sonuç oldukça doğrudur.